tests/component_handlers/schema_handler/test_schema_handler.py

```diff
@@ -96,79 +96,91 @@ def test_should_lazy_load_schema_provider(find_class_mock: MagicMock):
         schema_registry_url="http://localhost:8081",
     )
     schema_handler = SchemaHandler.load_schema_handler(
-        TEST_SCHEMA_PROVIDER_MODULE, config_enable
+        TEST_SCHEMA_PROVIDER_MODULE,
+        config_enable,
     )
 
     assert schema_handler is not None
 
     schema_handler.schema_provider.provide_schema(
-        "com.bakdata.kpops.test.SchemaHandlerTest", {}
+        "com.bakdata.kpops.test.SchemaHandlerTest",
+        {},
     )
     schema_handler.schema_provider.provide_schema(
-        "com.bakdata.kpops.test.SomeOtherSchemaClass", {}
+        "com.bakdata.kpops.test.SomeOtherSchemaClass",
+        {},
     )
 
     find_class_mock.assert_called_once_with(TEST_SCHEMA_PROVIDER_MODULE, SchemaProvider)
 
 
 def test_should_raise_value_error_if_schema_provider_class_not_found():
     schema_handler = SchemaHandler(
-        url="http://mock:8081", components_module=NON_EXISTING_PROVIDER_MODULE
+        url="http://mock:8081",
+        components_module=NON_EXISTING_PROVIDER_MODULE,
     )
 
-    with pytest.raises(ValueError) as value_error:
+    with pytest.raises(
+        ValueError,
+        match="No schema provider found in components module pydantic.main. "
+        "Please implement the abstract method in "
+        f"{SchemaProvider.__module__}.{SchemaProvider.__name__}.",
+    ):
         schema_handler.schema_provider.provide_schema(
-            "com.bakdata.kpops.test.SchemaHandlerTest", {}
+            "com.bakdata.kpops.test.SchemaHandlerTest",
+            {},
         )
 
-    assert (
-        str(value_error.value)
-        == "No schema provider found in components module pydantic.main. "
-        "Please implement the abstract method in "
-        f"{SchemaProvider.__module__}.{SchemaProvider.__name__}."
-    )
-
 
-def test_should_raise_value_error_when_schema_provider_is_called_and_components_module_is_empty():
+@pytest.mark.parametrize(
+    ("components_module"),
+    [
+        pytest.param(
+            None,
+            id="components_module = None",
+        ),
+        pytest.param(
+            "",
+            id="components_module = ''",
+        ),
+    ],
+)
+def test_should_raise_value_error_when_schema_provider_is_called_and_components_module_is_empty(
+    components_module,
```

Comment:
is there a rule to mark untyped parameters as warning?

Reply:
`ANN`

Reply:
ah ok, sorry I just saw your comment about it

Reply:
otherwise I can be the one to enforce this rule manually 🤣

Reply:
alternatively, we could configure Pyright to warn on symbols where it needs a type annotation, i.e. in all cases where it can't be inferred.

Reply:
:thinking: That's quite nice, it is probably the better approach imo

Reply:
Outright enabling `strict` mode produced 999 errors, 
```toml
[tool.pyright]
strict = ["*"]
```
while enabling just the following 5 rules produced 849:
```toml
[tool.pyright]
reportUnknownArgumentType = "error"
reportUnknownLambdaType = "error"
reportUnknownMemberType = "error"
reportUnknownParameterType = "error"
reportUnknownVariableType = "error"
```

[Pyright configuration docs](https://microsoft.github.io/pyright/#/configuration)

Reply:
#367